runelite-client/src/main/java/net/runelite/client/plugins/npchighlight/NpcIndicatorsPlugin.java

```diff
@@ -257,14 +257,25 @@ public void onMenuEntryAdded(MenuEntryAdded event)
 		{
 			type -= MENU_ACTION_DEPRIORITIZE_OFFSET;
 		}
-
-		if (config.highlightMenuNames() &&
-			NPC_MENU_ACTIONS.contains(MenuAction.of(type)) &&
-			highlightedNpcs.stream().anyMatch(npc -> npc.getIndex() == event.getIdentifier()))
+		if (NPC_MENU_ACTIONS.contains(MenuAction.of(type)))
```

Comment:
You can combine these if statements then check `isDead && recolorDeadNPCs` when the color is set

Reply:
That would make the dead NPC recoloring rely on the NPC already being highlighted if I'm interpretting you correctly, which doesn't seem to make sense to me. We could add an option to only recolour the names of dead highlighted NPCs, but I don't see how that has any extra use over just recolouring _all_ dead NPCs.